### PR TITLE
Adds a Bunch of New Religious Clothing Items

### PR DESCRIPTION
**Pull request by Farquaar**
## About The Pull Request
This PR adds several new clothing items used in religious practice around the world, including the: 
- [Kippah](https://en.wikipedia.org/wiki/Kippah)
- [Taqiyah ](https://en.wikipedia.org/wiki/Taqiyah_(cap))(red and white variants) 
- [Monk's frock](https://en.wikipedia.org/wiki/Religious_habit) (toggleable hood included)
- [Eastern monk robe](https://en.wikipedia.org/wiki/Kasaya_(clothing))
- [White robe](https://en.wikipedia.org/wiki/Thawb)
- [Rastacap/beanie](https://en.wikipedia.org/wiki/Rastacap)
- Bishop robes and [mitre](https://en.wikipedia.org/wiki/Mitre)`(premium chaplain item)
- [Clown priest](https://www.youtube.com/watch?v=zlwE5-z9ixw) [robes ](https://www.youtube.com/watch?v=43ObSAgnY1s)[and ](https://www.youtube.com/watch?v=HqIY3NfoYvw)[hat](https://aberrantkenosis.bandcamp.com/track/canon-in-h) (contraband for chaplain vendor, premium for autodrobe)
- [Medieval Jewish Hat](https://en.wikipedia.org/wiki/Jewish_hat) (contraband chaplain item)

I've added some preview images below. I've never contributed anything of this scale before, so any feedback on my coding and spritework is appreciated.

preview
monkpreview

Also, to cut down on repeated code, I created chaplainsuits.dm, which defines a subtype for chaplain/religious suit objects. It's basically defined as allowing general religious items in the suit storage slot (Bible, candles etc.) I put holiday priest and a couple of other existing in there, which required some updates to the pathing in a few places.

## Why It's Good For The Game
Given the breadth of different religions one might see on the station, the amount of clothing options for chaplains and followers is sorely lacking in variety. This should help players who want to roleplay as followers of non-Christian religions, while also spicing up the Christian chaplain roleplay meta.

Also, the addition of the new chaplainsuit subtype cuts down on repeated code.

## Changelog
:cl:
add: Added a number of Nanotrasen-approved religious clothing items to the chaplain vendor, autodrobe and clothesmate.
imageadd: sprites and icons for several new clothing items, including some in-hand sprites.
/:cl:

### CODE REVIEW

File: code/modules/clothing/head/beanie.dm [hunk 1]
```diff
@@ -89,5 +89,11 @@
 	desc = "If you're going on a worldwide hike, you'll need some cold protection."
 	icon_state = "waldo_hat"
 	item_color = "waldo_hat"
+	
+/obj/item/clothing/head/beanie/rasta
+	name = "rastacap"
+	desc = "Perfect for tucking in those dredlocks."
```
Comment (ATH1909) [inline code/modules/clothing/head/beanie.dm]: Shouldn't "dredlocks" be spelled as "dreadlocks"?

Comment (Farquaar) [reply code/modules/clothing/head/beanie.dm]: Good catch!

File: code/modules/clothing/suits/miscellaneous.dm [hunk 2]
```diff
@@ -177,6 +177,15 @@
...
 	allowed = list(/obj/item/storage/book/bible, /obj/item/nullrod, /obj/item/reagent_containers/food/drinks/bottle/holywater, /obj/item/storage/fancy/candle_box, /obj/item/candle, /obj/item/tank/internals/emergency_oxygen, /obj/item/tank/internals/plasmaman)
 
+/obj/item/clothing/suit/clownpriest
+	name = "Robes of the Honkmother"
+	desc = "Meant for a clown of the cloth."
+	icon_state = "clownpriest"
+	item_state = "clownpriest"
+	body_parts_covered = CHEST|GROIN|LEGS|ARMS
+	flags_inv = HIDEJUMPSUIT
+	allowed = list(/obj/item/storage/book/bible, /obj/item/nullrod, /obj/item/reagent_containers/food/drinks/bottle/holywater, /obj/item/storage/fancy/candle_box, /obj/item/candle, /obj/item/tank/internals/emergency_oxygen, /obj/item/tank/internals/plasmaman)
```
Comment (ATH1909) [inline code/modules/clothing/suits/miscellaneous.dm]: You should add bike horns, bananas, banana peels, water flowers, and other clown-related items to this list.

**Comment by subject217** [comment]
This is a great PR but /tg/ station will probably use these near exclusively for roleplaying as exaggerated racial stereotypes.

**Comment by HomocidlePancakes** [comment]
No Orthodoxian church clothes. How am I supposed to roleplay a russian drunk priest without them?

**Comment by anconfuzedrock** [comment]
Why is jew hat  illegal but not the mitre
I was gonna do this anyway

**Comment by bgobandit** [comment]
because our playerbase will probably use it for shitty anti-semitic crap so it has to take extra effort

**Comment by Farquaar** [comment]
While the medieval Jewish hat can have antisemitic connotations as it was occasionally forced upon Jews to distinguish them from the Christian or Muslim majority, it doesn't have to. I based the sprite after the hats of the [Birds' Head Haggadah](https://en.wikipedia.org/wiki/Birds%27_Head_Haggadah), an ancient Jewish manuscript that's filled with fun illustrations of birds celebrating Passover.

In all honesty though, the decision to include it was based solely on the fact that it is a silly hat, and one of my favourite hats of all time.

**Comment by anconfuzedrock** [comment]
also hijab when

**Comment by Farquaar** [comment]
Orthodox gear and hijabs are on the to-do list. I have some other contribs that are 90% done before I get on that, though.

**Comment by subject217** [comment]
>because our playerbase will probably use it for shitty anti-semitic crap so it has to take extra effort

This seems far more likely for the kippah than the "Jewish Hat".

If the /tg/ station playerbase can find a way to make it racist, antisemitic, islamophobic, what have you, they will. The question is really, *are there people who are interested in using this for its intended purpose?* I.e, increasing the flavor associated with traditional religions. I think there's definitely something there. Even if these are going to be racism accessories the other half of the time, that never stopped "Tyrone Watermelon" and "Schlomo Shekelberg" to start with.

**Comment by anconfuzedrock** [comment]
all I'm saying is when I inevitably make the islam quirk PR you may rest knowing that I would've made it regardless of if someone else added hijabs or I did it myself

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]
can you just subtype a suit for chaplainwear with the allowed var? just seeing the same thing over and over knowing they all have to be maintained should something get added/dropped hurts.

**Comment by Farquaar** [comment]
> 
> 
> can you just subtype a suit for chaplainwear with the allowed var? just seeing the same thing over and over knowing they all have to be maintained should something get added/dropped hurts.

That's a very good idea. I'll look up how to define a var and get back to you.
Where should I define the var, do you think?

**Comment by ExcessiveUseOfCobblestone** [comment]
make a dummy subtype like 
```DM
obj/item/clothing/suit/chaplainwear
    allowed = //the list you keep putting everywhere

obj/item/clothing/suit/hooded/chaplainwear // if you have multiple hooded
    allowed = //the list you keep putting everywhere
```

then have all your suits be subtyped to that so

```DM
obj/item/clothing/suit/chaplainwear/firstsuit
    blah blah blah
    // DO NOT ADD allowed

obj/item/clothing/suit/chaplainwear/secondsuit
    blah blah blah

**Comment by ExcessiveUseOfCobblestone** [comment]
yummy yummy webbits

**Comment by RaveRadbury** [comment]
You could probably toss the hats into the chaplainsuits.dm as well

**Comment by Farquaar** [comment]
AppVeyor is telling me that the type path is undefined. I'm still trying to figure where the problem is (I defined my subtype and actual items), but any tips in the right direction would be appreciated.

**Comment by Farquaar** [comment]
> 
> 
> You could probably toss the hats into the chaplainsuits.dm as well
I'm hesitant to do so, as the hats are, well, hats. I don't intend to make chaplainhats.dm right now, but maybe in the future.

**Comment by RaveRadbury** [comment]
You're already including hoods. This is just a way to keep things organized so you're clearly indicating that x hat goes with x suit.

**Comment by Farquaar** [comment]
Alright, I think I've hit the limit of what I understand coding-wise. Not quite sure what the Travis CI does.

> 
> 
> You're already including hoods. This is just a way to keep things organized so you're clearly indicating that x hat goes with x suit.

From what I've noticed, toggleable hoods are kept in the same .dm file as the suit (winter coats for instance), while separate hats (i.e. nun hood) are kept in the appropriate head .dm file. I'm just following convention atm

**Comment by ExcessiveUseOfCobblestone** [comment]
you can't webedit a file addition

**Comment by Farquaar** [comment]
> 
> 
> you can't webedit a file addition

Which one did I do wrong?

**Comment by Tlaltecuhtli** [comment]
make stuff like hijabs craftable/duplicable for rp purposes

**Comment by ExcessiveUseOfCobblestone** [comment]
> > you can't webedit a file addition
> 
> Which one did I do wrong? Any advice on how to fix it? I think I may have reached the limit of my meager knowledge.

you have to do it (aka tick the box beside the new file) locally and compile so it's added to the dme.

**Comment by Farquaar** [comment]
I think this is going BYOND my level of expertise here. Should I just delete the new chaplainsuits.dm and put all the new suits and subtypes in suits\jobs.dm?

**Comment by RaveRadbury** [comment]
@ExcessiveUseOfCobblestone I tried checking the box and compiling, git registered no changes and it wouldn't let me commit. Shouldn't having the .dme altered (which it is in this PR) do the trick? Or, how does one check that box and have git recognize it?

**Comment by ExcessiveUseOfCobblestone** [comment]
once you check the box and compile, the DME will also show a 1line change (the new file's path) and you will need to commit that unless it's changed very recently.

**Pull request merged**